packages/domain-picker/src/domain-picker/index.tsx

```diff
@@ -89,69 +83,43 @@ const DomainPicker: FunctionComponent< Props > = ( {
 	const [ domainSearch, setDomainSearch ] = useState< string >( initialDomainSearch );
 	const [ domainCategory, setDomainCategory ] = useState< string | undefined >();
 
-	const domainSuggestions = useDomainSuggestions(
+	const allDomainSuggestions = useDomainSuggestions(
 		domainSearch.trim(),
 		quantityExpanded,
 		domainCategory,
 		useI18n().i18nLocale
 	);
 
-	// Reset expansion state after every search
-	useEffect( () => {
-		setIsExpanded( false );
-	}, [ domainSearch ] );
-
-	const allSuggestions = domainSuggestions;
-	const freeSuggestions = getFreeDomainSuggestions( allSuggestions );
-	const paidSuggestions = getPaidDomainSuggestions( allSuggestions )?.slice(
+	const domainSuggestions = allDomainSuggestions?.slice(
 		0,
 		isExpanded ? quantityExpanded : quantity
 	);
-	const recommendedSuggestion = getRecommendedDomainSuggestion( paidSuggestions );
-
-	const paidSuggestionsWithoutRecommended = paidSuggestions?.filter(
-		( suggestion ) => suggestion !== recommendedSuggestion
-	);
-
-	// We're not using auto-selection right now.
-	// useEffect( () => {
-	// 	// Auto-select one of the domains when the search results change. If the currently
-	// 	// confirmed domain is in the search results then select it. The user probably
-	// 	// re-ran their previous query. Otherwise select the free domain suggestion.
-
-	// 	if (
-	// 		allSuggestions?.find(
-	// 			( suggestion ) => currentDomain?.domain_name === suggestion.domain_name
-	// 		)
-	// 	) {
-	// 		setCurrentSelection( currentDomain );
-	// 		return;
-	// 	}
 
-	// 	// Recalculate free-domain suggestions inside the closure. `getFreeDomainSuggestions()`
-	// 	// always returns a new object so it shouldn't be used in `useEffects()` dependencies list.
-	// 	const latestFreeSuggestion = getFreeDomainSuggestions( allSuggestions );
+	// the API puts the subdomain on top, push it to second place
+	if ( domainSuggestions && domainSuggestions.length > 1 ) {
+		domainSuggestions?.splice( 0, 2, domainSuggestions[ 1 ], domainSuggestions[ 0 ] );
```

Comment:
This feels fragile to me. I'd still prefer to have this extract the free options using `is_free` and then re-insert those results after the first position.

Reply:
Yes I changed it to that. Thanks!